Add support (to microdnf) for xml:base attribute from primary.xml (RhBug:1691315)

kontura opened the pull request:
xml:base attribute has priority over repository location.

The second commit discards possible file:// protocol from filename, because it doesn't work with it, for example g_file_test(...) can't handle filename with file:// protocol.

adelton commented on libdnf/dnf-package.cpp:
```diff
@@ -131,6 +131,13 @@ dnf_package_get_filename(DnfPackage *pkg)
     }
 
+    /* remove file:// protocol */
+    if (g_str_has_prefix(priv->filename, "file://")){
```
Similar to https://github.com/rpm-software-management/createrepo_c/pull/124, the check likely should be for `file:///`.

j-mracek commented:
@rh-atomic-bot try

j-mracek commented on libdnf/dnf-package.cpp:
```diff
@@ -113,9 +113,14 @@ dnf_package_get_filename(DnfPackage *pkg)
+            priv->filename = g_build_filename(url_location,
                                               dnf_package_get_location(pkg),
                                               NULL);
+            g_free(url_location);
```
I believe that you don't need to make g_strdup.

kontura commented:
I removed unnecessary memory allocation for `url_location` string and added the additional '/'. 

The additional '/' from what I understand prevents from returning filename without '/' as a first character for example from "file://host.example.com/path/to/file". But this way the example "file://host.example.com/path/to/file" would return with the file protocol still present and the subsequent g_file_test would fail. 
Question is how much is this relevant, because it isn't possible to download files over the network like this (or is it?). But "file://localhost/path/to/file" should work?

adelton commented:
My understanding is that `file:/path/here` and `file:///path/here` should both result in `/path/here`. The `file://machine/path/here` should likely fail loudly because the operation over network is not supported. As for `file://localhost/path/here`, technically it's an instance of the `file://machine/path/here` case so there is no problem if that fails as well. After all, people may have `/etc/hosts` configured to point `localhost` to some remote location for whatever reason ...

kontura commented:
Yes, that makes sense to me. Thank you for the help.
I added the code for `file:/path/here` use case and I also added error propagation for `file://machine/path/here` case.

j-mracek commented on libdnf/dnf-package.h:
```diff
@@ -60,7 +60,8 @@ typedef enum {
                                                          DnfRepo        *repo);
-const gchar     *dnf_package_get_filename               (DnfPackage     *pkg);
+const gchar     *dnf_package_get_filename               (DnfPackage     *pkg,
+                                                         GError         **errror);
```
It breaks API.

Conan-Kudo replied:
Also, `errror` -> `error`
